hudi-cli/src/main/java/org/apache/hudi/cli/commands/TableCommand.java

```diff
@@ -233,10 +237,28 @@ public String deleteTableConfig(
     return renderOldNewProps(newProps, oldProps);
   }
 
-  @ShellMethod(key = "table change-table-type", value = "Change hudi table type, COW_TO_MOR or MOR_TO_COW.")
+  @ShellMethod(key = "table change-table-type", value = "Change hudi table type to target type: COW or MOR. "
+      + "Note: before changing to COW, by default this command will execute all the pending compactions and execute a full compaction if needed.")
   public String changeTableType(
-      @ShellOption(value = {"--type"},
-          help = "COW_TO_MOR or MOR_TO_COW stands for changing table from COW to MOR or from MOR to COW respectively.") final String changeType) {
+      @ShellOption(value = {"--target-type"},
+          help = "the target hoodie table type: COW or MOR") final String changeType,
+      @ShellOption(value = {"--enable-compaction"}, defaultValue = "true",
+          help = "Valid in MOR to COW case. Before changing to COW, need to perform a full compaction to compact all log files. Default true") final boolean enableCompaction,
+      @ShellOption(value = {"--parallelism"}, defaultValue = "3",
+          help = "Valid in MOR to COW case. Parallelism for hoodie compaction") final String parallelism,
+      @ShellOption(value = "--schemaFilePath", defaultValue = "",
+          help = "Valid in MOR to COW case. Path for Avro schema file") final String schemaFilePath,
```

Comment:
Maybe there is no need for schema string, it can be deduced from the table schema I guess.

Reply:
Those params are copied from CompactionCommand::compact. Maybe it can be used in some edge case?

Reply:
It can be used as a fallback, the `--schemaFilePath` is awkward to use.